src/osd/scrubber/pg_scrubber.cc

```diff
@@ -775,7 +775,14 @@ bool PgScrubber::range_intersects_scrub(const hobject_t& start,
 
 Scrub::BlockedRangeWarning PgScrubber::acquire_blocked_alarm()
 {
-  ceph::timespan grace_period{m_debug_blockrange?4s:120s};
+  int grace = get_pg_cct()->_conf->osd_blocked_scrub_grace_period;
+  if (grace == 0) {
+    // we will not be sending any alarms re the blocked object
+    dout(20) << __func__ << ": blocked-alarm disabled" << dendl;
+    return nullptr;
+  }
+  ceph::timespan grace_period{m_debug_blockrange?4s:seconds{grace}};
+  dout(30) << __func__ << ": timeout:" << grace_period.count() << dendl;
```

Comment:
Nit: Add spaces on line 784 for readability:
`ceph::timespan grace_period{m_debug_blockrange ? 4s : seconds{grace}};`

Nit: add space after `timeout:`:
dout(30) << __func__ << ": timeout: " << grace_period.count() << dendl;